projects/baby_driver/scripts/test_i2c_write.py

```diff
@@ -0,0 +1,174 @@
+
+        i2c_write(1, 10, [7, 5, 12, 14, 15, 18, 9], 1)
+        self.assertEqual(9, self.babydriver_id)
+        self.assertEqual([7, 5, 12, 14, 15, 18, 9], self.data)
+        self.assertEqual(None, self.channel)
+        self.assertEqual(BABYDRIVER_CAN_MESSAGE_ID, self.msg_id)
+        self.assertEqual(BABYDRIVER_DEVICE_ID, self.device_id)
+
+    @patch('can_util.send_message')
+    @patch('can_util.next_message')
+    def test_send_message_max(self, mock_next_message, mock_send_message):
+        """Tests accuracy of minimum parameters passed into can_util.send_message"""
+        # Stores parameters passed into can_util.send_message
+        # pylint: disable=attribute-defined-outside-init
+        self.babydriver_id = None
+        self.data = None
+        self.channel = None
+        self.msg_id = None
+        self.device_id = None
+
+        def test_parameter(
+            babydriver_id=None,
+            data=None,
+            channel=None,
+            msg_id=BABYDRIVER_CAN_MESSAGE_ID,
+            device_id=BABYDRIVER_DEVICE_ID,
+            ):
+            self.babydriver_id = babydriver_id
+            self.data = data
+            self.channel = channel
+            self.msg_id = msg_id
+            self.device_id = device_id
+
+            mock_send_message.side_effect = test_parameter
+            mock_next_message.return_value.data = [0,0]
+
+            # Tests high parameters for can_util.send_message
+            i2c_write(1, 10, [7, 5, 12, 7, 5, 12, 8, 1, 2, 3, 4, 5, 6, 7], 1)
+            self.assertEqual(9, self.babydriver_id)
+            self.assertEqual([1, 2, 3, 4, 5, 6, 7], self.data)
+            self.assertEqual(None, self.channel)
+            self.assertEqual(BABYDRIVER_CAN_MESSAGE_ID, self.msg_id)
+            self.assertEqual(BABYDRIVER_DEVICE_ID, self.device_id)
+
+            i2c_write(1, 255, [7, 5, 12, 7, 5, 12, 7, 255, 255, 255, 255, 255, 255, 255], 1)
+            self.assertEqual(9, self.babydriver_id)
+            self.assertEqual([255, 255, 255, 255, 255, 255, 255], self.data)
+            self.assertEqual(None, self.channel)
+            self.assertEqual(BABYDRIVER_CAN_MESSAGE_ID, self.msg_id)
+            self.assertEqual(BABYDRIVER_DEVICE_ID, self.device_id)
+
+            i2c_write(1, 10, [255,255,255], 1)
+            self.assertEqual(9, self.babydriver_id)
+            self.assertEqual([255, 255, 255], self.data)
+            self.assertEqual(None, self.channel)
+            self.assertEqual(BABYDRIVER_CAN_MESSAGE_ID, self.msg_id)
+            self.assertEqual(BABYDRIVER_DEVICE_ID, self.device_id)
+
+            i2c_write(1, 10, [7, 5, 12], 255)
+            self.assertEqual(9, self.babydriver_id)
+            self.assertEqual([7, 5, 12], self.data)
+            self.assertEqual(None, self.channel)
+            self.assertEqual(BABYDRIVER_CAN_MESSAGE_ID, self.msg_id)
+            self.assertEqual(BABYDRIVER_DEVICE_ID, self.device_id)
```

Comment:
Because of the indentation here, all of these asserts are actually inside the `test_parameter` function, so they aren't run. You'll need to deindent them so they run.